src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution_ArgsToParameters.cs

```diff
@@ -85,13 +85,12 @@ public ImmutableArray<int> ToImmutableArray()
 
             bool seenNamedParams = false;
             bool seenOutOfPositionNamedArgument = false;
-            bool isValidParams = IsValidParams(symbol);
```

Comment:
Is this validation performed elsewhere, or why is it OK to remove it here?

Reply:
> Is this validation performed elsewhere, or why is it OK to remove it here?

I am using the following line of thinking. If we are not dealing with expanded form, it shouldn't matter whether there is a valid `params` parameter. Even if one is there, a normal form is supposed to use it as a regular parameter. If we are dealing with expanded form, we already performed all necessary checks and decided that there is a valid `params` parameter. Rechecking the fact again is redundant. Not only that, it actually might yield an unexpected value, because the fact whether a candidate should be considered in expanded form is derived from the least overridden member. This code uses the most overridden member. So, theoretically, this check can produce `false` even if we already decided to analyze an expanded form. See [https://github.com/dotnet/roslyn/issues/70908](https://github.com/dotnet/roslyn/issues/70908), for example.